packages/web-components/src/components/search/search.ts

```diff
@@ -76,13 +76,73 @@ class CDSSearch extends HostListenerMixin(FocusMixin(FormMixin(LitElement))) {
   }
 
   /**
-   * Handles `focus` event on the button when the button can be expanded
+   * Expand only when the magnifier icon is clicked
    */
-  @HostListener('focus')
+  @HostListener('click')
   // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
-  private _handleExpand() {
-    if (this.expandable && !this.expanded) {
+  private _handleExpand(e: Event) {
+    // Check if the click came from the magnifier area
+    const path = (e.composedPath && (e.composedPath() as unknown[])) || [];
+    const isMagnifierClick = path.some((n: unknown) =>
+      (n as Element)?.classList?.contains(`${prefix}--search-magnifier`)
+    );
+
+    if (isMagnifierClick && this.expandable && !this.expanded) {
       this.setAttribute('expanded', '');
+      // Focus the input after expanding
+      this.shadowRoot!.getElementById('input')?.focus();
+    }
+  }
```

Comment:
```suggestion
    if (isMagnifierClick && this.expandable && !this.expanded) {
      this._expandAndFocus();
    }
  }
  
private _expandAndFocus() {
  this.setAttribute('expanded', '');
  // Focus the input after expanding
  this.shadowRoot?.getElementById('input')?.focus();
}
```

This block, as well as the last `if` in `_handleKeys` do the same thing. They can probably share a function to avoid repetition.